### DISCUSSION TITLE
add_line function is not drawing as per supplied x and y values

### DISCUSSION THREAD

**Discussion by deepspkd** [highfestiva/finplot#462] [Q&A] [answered]
I was trying to draw lines as per left mouse clicks. mouse clicks are identified using set_mouse_callback function and line is drawn as it receives left click and hover datas. code using which line is drawn is as follows.

    def mouseclick(self, x, y):

        if self.is_drawing_active == True:
            if self.is_left_mouse_clicked == True:
                self.draw_line_B_x.append(x)
                self.draw_line_B_y.append(y)
                self.is_left_mouse_clicked = None
                self.is_drawing_active = None
                self.line = None
                self.line_text = None

        else:
            self.is_drawing_active = True
            self.is_left_mouse_clicked = True
            self.draw_line_A_x.append(x)
            self.draw_line_A_y.append(y)

    def mousehover(self, x, y):
        if self.is_drawing_active == True:
            if self.is_left_mouse_clicked == True:
                if self.line:
                    fplt.remove_primitive(self.line)
                    fplt.remove_primitive(self.line_text)
                self.line = fplt.add_line((self.draw_line_A_x[-1],self.draw_line_A_y[-1]),(x,y),interactive=False)
                self.line_text = fplt.add_text((self.draw_line_A_x[-1],self.draw_line_A_y[-1]),"Line")

But i am not getting a line as per given x and y values. If the first x and y value (A part of the line AB) comes on the right side of the line, the drawn line respects the given y value but x value is changed somehow. Please help.

https://github.com/highfestiva/finplot/assets/66288861/f1d55258-f51f-4f26-82e0-73ff5a19faa3

**Comment by highfestiva** [comment]
Haha, interesting! I don't know, but I suppose you're getting multiple `mouseclick` callbacks. If you print your X and Y arrays, you're probably going to see them increase over time. Also it looks as if something strange is happening with your X-values, perhaps in some other piece of code. If you want more help, I'd need a complete, minimal example which I can run myself.

**Reply by deepspkd** [reply]
I checked for mistakes in X and Y values supplied and to see it visually i used the same X and Y value to draw the text "Line". But the text is drawn correctly but line is drawn incorrectly. 

A minimal example is attached. 
[db.zip](https://github.com/highfestiva/finplot/files/13459366/db.zip)

**Reply by highfestiva** [reply, marked as answer]
You're right, thanks for the example and the bug report. Fix in 26de5e8.

**Comment by deepspkd** [comment]
Thankyou

**Reply by kumarangp** [reply]
application you are shown in the video is awesome! Can you explain steps to create a application like that?

**Comment by deepspkd** [comment]
I am using PyQt6 for GUI. Jugaad Data for bhavcopy downloading. finplot for all drawing operations and Pandas for bhav data processing. I learned basics of PyQt and Pandas from youtube, library documentation and other online resources, finplot using examples, issues, and discussions. Then using my learnings to visualise my ideas in to charts using this awesome library.
